appserver/ejb/ejb-container/src/main/java/org/glassfish/ejb/security/application/EJBSecurityManager.java

```diff
@@ -785,21 +359,19 @@ public Principal getCallerPrincipal() {
     @Override
     public void destroy() {
         try {
-            boolean wasInService = getPolicyFactory().inService(this.contextId);
-            if (wasInService) {
-                policy.refresh();
-            }
+            authorizationService.refresh();
+
             /*
-             * all ejbs of module share same policy context, but each has its
+             * All enterprise beans of module share same policy context, but each has its
              * own permission cache, which must be unregistered from factory to
-             * avoid leak.
+             * avoid leaks.
              */
             PermissionCacheFactory.removePermissionCache(uncheckedMethodPermissionCache);
             uncheckedMethodPermissionCache = null;
-            roleMapperFactory.removeAppNameForContext(this.contextId);
+            roleMapperFactory.removeAppNameForContext(contextId);
 
-        } catch (PolicyContextException pce) {
```

Comment:
This smells, catching just ISE? What about other RTE? They should or should not be propagated? At least comment why is this catch block so special would be fine.

Reply:
The previous code caught PolicyContextException as a special case. This has been replaced by an IllegalStateException. It just maintains that behaviour.

Reply:
Hmm, I don't like it much, but ok.